### PR TITLE
Add command block coverage merging

### PR DESCRIPTION
**Pull request by ranweiler**
- Add coverage merging for module and command-level block coverage, tests
- Improve an error message

### CODE REVIEW

File: src/agent/coverage/src/block.rs [hunk 1]
```diff
@@ -40,12 +40,10 @@ impl CommandBlockCov {
 
     pub fn increment(&mut self, path: &ModulePath, offset: u64) -> Result<()> {
         if let Some(module) = self.modules.get_mut(path) {
```
Comment (ranweiler) [inline src/agent/coverage/src/block.rs:42]: Now, when we _merge_ command coverage, we now add unknown modules to the overall coverage. Here, we drop increment attempts for unknown modules. The reason this changed is that although right now we expect to precompute possible coverage locations, we'd like this format to support coverage discovery.

We should probably create a new (empty) `ModuleCov` for `path` as-needed.

File: src/agent/coverage/src/block.rs [hunk 2]
```diff
@@ -69,6 +74,22 @@ impl ModuleCov {
         let blocks = offsets.map(|o| (o, BlockCov::new(o))).collect();
         Self { blocks }
     }
+
+    pub fn increment(&mut self, offset: u64) {
+        if let Some(block) = self.blocks.get_mut(&offset) {
```
Comment (ranweiler) [inline src/agent/coverage/src/block.rs:79]: Again, we should probably create an entry on demand.

File: src/agent/coverage/src/block.rs [hunk 3]
```diff
@@ -95,3 +118,151 @@ impl BlockCov {
...
+        cov
+    }
+
+    // Builds a vec of `(count, offset)` tuples from a `ModuleCov`.
+    fn to_vec(cov: &ModuleCov) -> Vec<(u64, u32)> {
+        cov.blocks.iter().map(|(o, b)| (*o, b.count)).collect()
+    }
+
+    #[test]
+    fn test_to_from_vec_roundtrip() {
```
Comment (lzybkr) [inline src/agent/coverage/src/block.rs:144]: Based on the test name and code, it looks like it only tests test code, but I think it's meant to test `increment` so maybe a different function name would be better?

Comment (ranweiler) [reply src/agent/coverage/src/block.rs:144]: Oops, nope, this is just a vestigial self-check test.

**Review by lzybkr — APPROVED** [review]

**Pull request merged**